### PR TITLE
Linked licences fusioncode

### PR DESCRIPTION
**Pull request by jjaumotte**
Pourrais-tu me faire part de tes remarques/améliorations ? Voici le ticket lié : https://support.imio.be/browse/MURBECAA-52
Merci.

### CODE REVIEW

File: src/Products/urban/docgen/helper_view.py [hunk 1]
```diff
@@ -515,6 +516,32 @@ def _get_last_opinions(self, field_name):
...
+                        licences_limit_date.append(licence)
+                    elif delivered.getDecision() == decision_event_status:
+                        licences_limit_date.append(licence)
+            related_licences = licences_limit_date
+
+        if licence_status:
+            licences_status = []
+            for licence in related_licences:
+                workflow = getToolByName(licence, 'portal_workflow')
+                licence_review_state = workflow.getInfoFor(licence, 'review_state')
```
Comment (sdelcourt) [inline src/Products/urban/docgen/helper_view.py:538]: use plone.api instead

File: src/Products/urban/docgen/helper_view.py [hunk 2]
```diff
@@ -515,6 +516,32 @@ def _get_last_opinions(self, field_name):
             return opinions
         return []
 
+    def get_related_licences(self, licence_types=[], decision_limit_date=None, decision_event_status='', licence_status=None):
```
Comment (sdelcourt) [inline src/Products/urban/docgen/helper_view.py:519]: use 'state' rather than 'status'

File: src/Products/urban/docgen/helper_view.py [hunk 3]
```diff
@@ -515,6 +516,32 @@ def _get_last_opinions(self, field_name):
...
+                delivered = licence.getLastTheLicence()
+                if delivered and (delivered.getDecisionDate() or delivered.getEventDate()) >= decision_limit_date:
+                    if not decision_event_status:
+                        licences_limit_date.append(licence)
+                    elif delivered.getDecision() == decision_event_status:
+                        licences_limit_date.append(licence)
+            related_licences = licences_limit_date
+
+        if licence_status:
+            licences_status = []
```
Comment (sdelcourt) [inline src/Products/urban/docgen/helper_view.py:535]: use something more different from licence_status

**Review by sdelcourt — CHANGES_REQUESTED** [review]
Add a unit test ;-)

File: src/Products/urban/docgen/helper_view.py (hunk 3 above)
Comment (jjaumotte) [reply src/Products/urban/docgen/helper_view.py]: Fix variable name

File: src/Products/urban/docgen/helper_view.py (hunk 2 above)
Comment (jjaumotte) [reply src/Products/urban/docgen/helper_view.py]: Fix variable name

**Comment by jjaumotte** [comment]
Add unit tests for "with_historic" param

**Review by sdelcourt — APPROVED** [review]
all good (-:

**Pull request merged**